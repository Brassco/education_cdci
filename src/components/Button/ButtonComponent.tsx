import React from "react";

//Types
export type ButtonProps = {
	title: string;
	onPress: () => void;
};

//View
import ButtonView from "./ButtonView";

const ButtonComponent = ({title, onPress}: ButtonProps) => {
	return <ButtonView title={title} onPress={onPress} />;
};

export default ButtonComponent;
