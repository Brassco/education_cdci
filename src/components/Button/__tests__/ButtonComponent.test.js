import React from "react";
import {render, fireEvent} from "@testing-library/react-native";
import ButtonComponent from "../ButtonComponent";

describe("ButtonComponent", () => {
	test("renders with the correct title", () => {
		const title = "Click me";
		const {getByText} = render(
			<ButtonComponent title={title} onPress={() => {}} />
		);
		const buttonElement = getByText(title);
		expect(buttonElement).toBeDefined();
	});

	test("calls onPress when the button is pressed", () => {
		const onPressMock = jest.fn().mockReturnValue(1);
		const {getByText} = render(
			<ButtonComponent title="Click me" onPress={onPressMock} />
		);
		const buttonElement = getByText("Click me");

		fireEvent.press(buttonElement);

		// Check if onPress is called once
		expect(onPressMock).toHaveBeenCalledTimes(1);

		// If your onPress function returns a value, you can check it
		expect(onPressMock).toHaveReturnedWith(1);
	});
});
