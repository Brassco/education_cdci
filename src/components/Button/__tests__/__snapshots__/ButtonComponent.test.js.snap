// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderColor": "#188",
      "borderWidth": 1,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 5,
      "paddingVertical": 5,
    }
  }
>
  <Text
    style={
      {
        "color": "#1f1f1f",
        "fontSize": 18,
        "fontWeight": 600,
      }
    }
  >
    Login
  </Text>
</View>
`;
